src/ensembl/src/shared/search-field/SearchField.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { ReactNode } from 'react';
+import classNames from 'classnames';
+
+import styles from './SearchField.scss';
+
+import Input from 'src/shared/input/Input';
+
+type Props = {
+  search: string;
+  placeholder?: string;
+  onChange: (value: string) => void;
+  onSubmit?: (value: string) => void;
+  onFocus?: () => void;
+  onBlur?: () => void;
+  onKeyUp?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
+  onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
+  onKeyPress?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
+  rightCorner?: ReactNode;
+  className?: string;
+};
+
+const SearchField = (props: Props) => {
+  const { rightCorner } = props;
+  const className = classNames(styles.searchField, props.className);
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+    props.onSubmit && props.onSubmit(props.search);
```

Comment:
Submission of empty string might be useful when the user wants to clear the input field to reset the results, but on the other hand we are dispatching an action which doesn't do anything. 
What is your opinion about preventing submission of empty string, but at the same time having an alternative way to reset the results?

Reply:
This is a very good question, and the one that I do not have a clear answer to.  I did not make this choice consciously when I wrote this code, so it's a great opportunity to explore the options. 

On the one hand, the `SearchField` component, as well as the `AutocompleteSearchField` component, is entirely dependent upon its parent to store the string that goes into the input field. Upon every change of the input field, it calls the `onChange` handler that it receives from its parent. Which means that one way of clearing search results is just to delete the content of the input field, and then rely on the parent component to hide the search results when the search string is empty. So here's the alternative way of resetting the results.

On the other hand, the `onSubmit` handler offers a way of signalling to the parent that the user has made a clear decision (as opposed to the `onChanged` handler that fires at every intermediate step as well). So I am a bit hesitant to block this way of signalling when the field is empty.

Ultimately, it's up to the parent component to decide what to do when the search field is empty anyway. So I am leaning towards having the `onSubmit` fire even when the search field is empty, and then have the logic in the parent component for dealing with these events in appropriate manner.